README.md

```diff
@@ -4,45 +4,145 @@
 
-## Disclaimer
-
-This demo and its code are intended for demonstrative purposes only and are not
-designed for performance.
+This example illustrates the application of a QBoost formulation towards two
+illustrative data sets.
 
 ## Usage
 
-To run a demonstration using one of two provided sample data sets, run:
+The demonstration includes two sample data sets, which can be run with the
+following commands:
 
 ```bash
-python demo.py  wisc
+python demo.py blobs
 ```
 
 Or:
 
 ```bash
-python demo.py mnist
+python demo.py digits
+```
+
+### Blobs data set
+
+This is a simple synthetically generated data set for illustrative purposes.  It
+is based on the scikit-learn function `sklearn.datasets.make_blobs`.  It
+generates a data set for classification with an arbitrary number of features,
+samples, and informative features.  The features are randomly generated from
+Gaussian distributions.  For each informative feature, the mean value differs
+between the two target classes, while the non-informative features are sampled
+from the same probability distribution in each class.
+
+Each run of the demo will randomly generate the data set and split it into
+training, validation, and test sets.  The validation set is used to determine
+the best value of the regularization parameter using a simple parameter sweep.
+The results of the demo will print the indices of the actual informative
+features, the features that are selected by QBoost, and the accuracy of the
+resulting ensemble classifier on the test set.
+
+The values of the number of samples, number of features, and number of
+informative features can be controlled using command line arguments, as
+described in the help:
+
+```bash
+python demo.py blobs -h
+```
+
+### Digits data set
+
+The digits data is based on the well-known handwritten digits data set.  Each
+instance is a digitized image of a handwritten digit.  The images consist of 8x8
+grayscale pixel values, which are represented as 64 features.  The goal is to
+construct a classifier that can predict the digit that is represented by an
+image.
+
+This demonstration constructs a binary classification problem using images for any two
+of the available ten digits, as specified by the `--digit1` and `--digit2`
+command line options (the defaults are 0 and 1).  As with the blobs data set,
+the available data are first split into training, validation, and test sets.
+The validation set is used to determine the value of the regularization
+parameter using a parameter sweep.  The demonstration will print the number of
+features selected by QBoost, along with the prediction accuracy on the test set.
+
+The `--plot-digits` option is also provided to display one instance of the
+images for each of the two selected digits.  The displayed images are chosen
+randomly from the available data.
+
+The following command displays the available options for use with the digits
+data set:
+
+```bash
+python demo.py digits -h
 ```
 
+## Code Overview
+
```

Comment:
> 
> 
> The content of this Code Overview is actually an expansion of introduction section (before Usage). It does not really give any code details.

I would say this section is a mix of a description of the method and the particular implementation used in the example code.  "QBoost" is somewhat of a general term, so I have used this section to highlight some of the specific choices made in this code, such as treatment of the weights with a bit depth of 1 (does not have to be done this way) and scaling of the weak classifier outputs.

Perhaps a better title would be "Implementation Overview", "Method Overview", or "Overview of QBoost Method"?  Interested if you have any suggestions.

> 
> This could be the place to say something about finding the best lambda, for example, because someone just running the code sees that output and it's not self-explanatory.

I went ahead and removed display of the lambda value from the output, as I don't think it provided any value for the user.  The README does still mention that a parameter sweep is used to determine lambda.

Reply:
I think this title is for consistency with other examples and the template. You've restructured the code so it reads much better now and I assume the flow is understandable to users with ML experience.